### PR TITLE
Dont use bitcode as it's now deprecated

### PR DESCRIPTION
**Pull request by rgaudin**
See https://stackoverflow.com/questions/72543728/xcode-14-deprecates-bitcode-but-why

Fixes https://github.com/openzim/libzim/issues/834

### CODE REVIEW

**Review by kelson42 — APPROVED** [review]
Lgtm

**Comment by kelson42** [comment]
@rgaudin But CI fails :(

**Comment by rgaudin** [comment]
OK the issue is inside the deps file that is downloaded from tmp.kiwix.org: meson_cross_file.txt includes `-fembed-bitcode`. I'll come back to this PR once this gets updated.

btw, those are not versioned 🤷‍♂️

**Comment by kelson42** [comment]
@mgautierfr We need to get something clean around this! See as well https://github.com/openzim/libzim/issues/834

**Comment by mgautierfr** [comment]
> OK the issue is inside the deps file that is downloaded from tmp.kiwix.org: meson_cross_file.txt includes -fembed-bitcode.

The deps file is generated everynigth from kiwix-build main. As long as https://github.com/kiwix/kiwix-build/pull/650 is not merged, `meson_cross_file.txt` will includes `-fembed-bicode`.

I have added a commit (to remove before merge) which use the dev_preview.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -12,7 +12,6 @@ jobs:
       fail-fast: false
       matrix:
         os:
-          - macos-11
           - macos-12
           - macos-13
           - macos-13-xlarge
```
Comment (mgautierfr) [inline .github/workflows/ci.yml:17]: I'm a bit afraid of that.
kiwix-build generate dependencies build for macos-13 but publish them using the name `macos` only (without version).

Should we removed macos-12 ?
If not, it would be better to make kiwix-build also generate macos-12 dependencies and add the mac-os version to the archive name.

Comment (kelson42) [reply .github/workflows/ci.yml:17]: > I'm a bit afraid of that. kiwix-build generate dependencies build for macos-13 but publish them using the name `macos` only (without version).
> 
> Should we removed macos-12 ? If not, it would be better to make kiwix-build also generate macos-12 dependencies and add the mac-os version to the archive name.
> 
> Edit: Well, ios CI is actually failing because `meson_cross_file.txt` points to compiler not existing in macos-12. We have to do something here:
> 
>     * Either build IOS on macos-13 (and drop macos-12)
> 
>     * Or build deps on macos-12 too.

@mgautierfr We could and [should remove macos-12 here and compile only on `macos-13`](https://github.com/kiwix/apple/pull/533), BUT then we have to put the right compilation parameter to secure macOS12 will be supported at runtime.

Comment (mgautierfr) [reply .github/workflows/ci.yml:17]: In https://github.com/kiwix/kiwix-build/pull/650 we set `-mmacosx-version-min` to `12.0` so we should be ok.

Comment (kelson42) [reply .github/workflows/ci.yml]: We are good there... but here?

Comment (kelson42) [reply .github/workflows/ci.yml]: > Well, ios CI is actually failing because `meson_cross_file.txt` points to compiler not existing in macos-12. We have to do something here:
> 
>     * Either build IOS on macos-13 (and drop macos-12)
> 
>     * Or build deps on macos-12 too.

This is https://github.com/openzim/libzim/issues/834. I guess moving to macos13 resolves the problem?! :)

**Comment by kelson42** [comment]
@rgaudin @mgautierfr CI is green, so does that mean this PR is ready to merge, just blocker by merging https://github.com/kiwix/kiwix-build/pull/650?

**Comment by mgautierfr** [comment]
On my side, mostly yes:
- Maybe reword commits (remove the `?` in `upgrade meson?`)
- Remove the last commit before merging !

**Comment by rgaudin** [comment]
@mgautierfr do we want to introduce `-mmacosx-version-min=12.0` here or do we consider that's a CD concern and that issues with this would circle-back?

I updated meson to lastest version instead as that's what we do every where.

I see you also force-pushed https://github.com/openzim/libzim/compare/44e4de62b2035f087b16443414b6f95fb0ebb72b..5dc4577ebb529228c89b013797d4bcaf34cc03ac and I don't know what to do with it.

Update the matrix as you see fit.

This could be merged D+1 after https://github.com/kiwix/kiwix-build/pull/650 is merged I think

**Comment by rgaudin** [comment]
@mgautierfr that's ready to merge
